File: internal/types/application.go
package types

import (
	"time"

	"github.com/google/uuid"
)

type Application struct {
	ID             uuid.UUID            `db:"id" json:"id"`
	CreatedAt      time.Time            `db:"created_at" json:"createdAt"`
	OrganizationID uuid.UUID            `db:"organization_id" json:"-"`
	Name           string               `db:"name" json:"name"`
	Type           DeploymentType       `db:"type" json:"type"`
	ImageID        *uuid.UUID           `db:"image_id" json:"imageId,omitempty"`
	Versions       []ApplicationVersion `db:"versions" json:"versions"`
}

File: sdk/js/src/types/application.ts
import {BaseModel, Named} from './base';
import {DeploymentType, HelmChartType} from './deployment';

export interface Application extends BaseModel, Named {
  type: DeploymentType;
  imageId?: string;
  imageUrl?: string;
  versions?: ApplicationVersion[];
}

export interface ApplicationVersion {
  id?: string;
  name: string;
  linkTemplate?: string;
  createdAt?: string;
  archivedAt?: string;
  applicationId?: string;
  chartType?: HelmChartType;
  chartName?: string;
  chartUrl?: string;
  chartVersion?: string;
  resources?: ApplicationVersionResource[];
}

export interface ApplicationVersionResource {
  id?: string;
  applicationVersionId?: string;
  name: string;
  content: string;
  visibleToCustomers: boolean;
}

export interface PatchApplicationRequest {
  name?: string;
  versions?: {id: string; archivedAt?: string}[];
}
